### DISCUSSION TITLE
Project Colin (test framework)

### DISCUSSION THREAD

**Discussion by dalers** [timschofield/webERP#272] [Development Discussion & Specification]
@timschofield thanks for pointing me to the [Colin test framework you created https://github.com/timschofield/colin](https://github.com/timschofield/colin). 

I'm trying to run it on the current project to see what error messages are generated, but I'm still sorting out path changes and including files correctly on my test server.

- Could Colin be used as-is on the current master branch? If not, why not? What generally would need doing?
- Would using Colin benefit v5 testing?
- Is the philosophy of Colin and the methods it uses still a good idea in your opinion? 

Has anyone else used Colin?

Thanks,
Dale

**Comment by dalers** [comment]
Fwiw...

I changed "php5" to simply "php" in the TestAll.php shebang. Not withstanding its' .php extension, I assume the shebang in TestAll.php means it can be executed from the shell, but I'm having trouble (which I expect will be a Doh! moment when I figure out what I'm doing wrong...)


```
dale@whizzer:/usr/local/www/colin $ ./TestAll.php
-sh: ./TestAll.php: not found
dale@whizzer:/usr/local/www/colin $ ls ./TestAll.php
./TestAll.php
dale@whizzer:/usr/local/www/colin $
```


```
dale@whizzer:/usr/local/www/colin $ ll
total 89
drwxrwxr-x   8 www www uarch    18 Dec  4 12:30 ./
drwxrwxr-x  28 www www uarch    30 Dec  4 12:30 ../
drwxrwxr-x   8 www www uarch    13 Dec  4 12:30 .git/
-rw-rw-r--   1 www www uarch    19 Dec  4 12:30 .gitignore
-rwxrwxr-x   1 www www uarch  3819 Dec  4 12:30 0.phpt*
-rwxrwxr-x   1 www www uarch  3619 Dec  4 12:30 1.phpt*
-rwxrwxr-x   1 www www uarch  1869 Dec  4 12:30 2.phpt*
-rwxrwxr-x   1 www www uarch  2298 Dec  4 12:30 3.phpt*
-rwxrwxr-x   1 www www uarch  2265 Dec  4 12:30 4.phpt*
-rwxrwxr-x   1 www www uarch  2285 Dec  4 12:30 5.phpt*
-rw-rw-r--   1 www www uarch 18027 Dec  4 12:30 LICENSE
-rw-rw-r--   1 www www uarch    39 Dec  4 12:30 README.md
-rwxrwxr-x   1 www www uarch   173 Dec  4 15:07 TestAll.php*
drwxrwxr-x   2 www www uarch     3 Dec  4 12:30 classes/
drwxrwxr-x   2 www www uarch     3 Dec  4 12:30 data/
drwxrwxr-x   3 www www uarch    10 Dec  4 12:30 includes/
drwxrwxr-x   2 www www uarch     6 Dec  4 12:30 public_html/
drwxrwxr-x   2 www www uarch     3 Dec  4 12:30 sql/
dale@whizzer:/usr/local/www/colin $
```


```
dale@whizzer:/usr/local/www/colin $ php --version
PHP 8.1.30 (cli) (built: Oct  8 2024 02:10:22) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.30, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.30, Copyright (c), by Zend Technologies
dale@whizzer:/usr/local/www/colin $
```


```
dale@whizzer:/usr/local/www/colin $ php ./TestAll.php
sh: /usr/local/www/colin/0.phpt: not found
127
sh: /usr/local/www/colin/1.phpt: not found
127
sh: /usr/local/www/colin/2.phpt: not found
127
sh: /usr/local/www/colin/3.phpt: not found
127
sh: /usr/local/www/colin/4.phpt: not found
127
sh: /usr/local/www/colin/5.phpt: not found
127
dale@whizzer:/usr/local/www/colin $
```

```
dale@whizzer:/usr/local/www/colin $ php ./0.phpt
PHP Fatal error:  Uncaught Error: Call to undefined function RegisterTest() in /usr/local/www/colin/0.phpt:9
Stack trace:
#0 {main}
  thrown in /usr/local/www/colin/0.phpt on line 9
dale@whizzer:/usr/local/www/colin $
```

**Comment by timschofield** [comment]
Hi Dale, I realised after that the version on GitHub was quite old. I have
updated it. However it is written to be run from the command line. You also
need to fill in config.dist.php and save it as config.php. On my system
here I install it in /home/tim/bin/colin and then I run it from that
directory. I would spend time understanding how the framework works.
If the mood inspires you, write something in README.md about how to get it
working.

Tim

On Wed, 4 Dec 2024 at 22:16, Dale Scott ***@***.***> wrote:

> Fwiw...
>
> I changed "php5" to simply "php" in the TestAll.php shebang. Not
> withstanding its' .php extension, I assume the shebang in TestAll.php means
> it can be executed from the shell, but I'm having trouble (which I expect
> will be a Doh! moment when I figure out what I'm doing wrong...)
>
> ***@***.***:/usr/local/www/colin $ ./TestAll.php
> -sh: ./TestAll.php: not found
> ***@***.***:/usr/local/www/colin $ ls ./TestAll.php
> ./TestAll.php
> ***@***.***:/usr/local/www/colin $
>
> ***@***.***:/usr/local/www/colin $ php --version
> PHP 8.1.30 (cli) (built: Oct  8 2024 02:10:22) (NTS)
> Copyright (c) The PHP Group
> Zend Engine v4.1.30, Copyright (c) Zend Technologies
>     with Zend OPcache v8.1.30, Copyright (c), by Zend Technologies
> ***@***.***:/usr/local/www/colin $
>
> ***@***.***:/usr/local/www/colin $ php ./TestAll.php
> sh: /usr/local/www/colin/0.phpt: not found
> 127
> sh: /usr/local/www/colin/1.phpt: not found
> 127
> sh: /usr/local/www/colin/2.phpt: not found
> 127
> sh: /usr/local/www/colin/3.phpt: not found
> 127
> sh: /usr/local/www/colin/4.phpt: not found
> 127
> sh: /usr/local/www/colin/5.phpt: not found
> 127
> ***@***.***:/usr/local/www/colin $
>
> ***@***.***:/usr/local/www/colin $ php ./0.phpt
> PHP Fatal error:  Uncaught Error: Call to undefined function RegisterTest() in /usr/local/www/colin/0.phpt:9
> Stack trace:
> #0 {main}
>   thrown in /usr/local/www/colin/0.phpt on line 9
> ***@***.***:/usr/local/www/colin $
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/272#discussioncomment-11466578>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LADSCPLJEFPOUNBOST2D55MRAVCNFSM6AAAAABTBB5BVOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNBWGY2TOOA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/
